src/fmu/dataio/_objectdata_provider.py

```diff
@@ -402,17 +402,17 @@ def _derive_spec_bbox_points(self) -> tuple[dict[str, Any], dict[str, Any]]:
 
         bbox: dict[str, Any] = {}
 
-        if len(pnts.dataframe.columns) > 3:
-            attrnames = pnts.dataframe.columns[3:]
+        if len(pnts.get_dataframe().columns) > 3:
+            attrnames = pnts.get_dataframe().columns[3:]
             specs["attributes"] = list(attrnames)
-        specs["size"] = int(pnts.dataframe.size)
-
-        bbox["xmin"] = float(pnts.dataframe[pnts.xname].min())
-        bbox["xmax"] = float(pnts.dataframe[pnts.xname].max())
-        bbox["ymax"] = float(pnts.dataframe[pnts.yname].min())
-        bbox["ymin"] = float(pnts.dataframe[pnts.yname].max())
-        bbox["zmin"] = float(pnts.dataframe[pnts.zname].min())
-        bbox["zmax"] = float(pnts.dataframe[pnts.zname].max())
+        specs["size"] = int(pnts.get_dataframe().size)
+
+        bbox["xmin"] = float(pnts.get_dataframe()[pnts.xname].min())
+        bbox["xmax"] = float(pnts.get_dataframe()[pnts.xname].max())
+        bbox["ymax"] = float(pnts.get_dataframe()[pnts.yname].min())
+        bbox["ymin"] = float(pnts.get_dataframe()[pnts.yname].max())
+        bbox["zmin"] = float(pnts.get_dataframe()[pnts.zname].min())
+        bbox["zmax"] = float(pnts.get_dataframe()[pnts.zname].max())
```

Comment:
If you onley need read-only from the dataframe, `get_dataframe(copy=False)` will return a view of the current dataframe; otherwise it will make a new copy which may pile up memory.